### PR TITLE
8367022: MethodHandles.foldArguments handle negative pos

### PR DESCRIPTION
**Pull request by benshuai5D**
Please review this small fix.

**Problem:**

`MethodHandles.foldArguments` throws `ArrayIndexOutOfBoundsException` instead of `IllegalArgumentException` for negative `pos` values, which violates the API's exception contract. Extreme values such as `Integer.MAX_VALUE` can also overflow the existing addition-based bounds check.

**Fix:**

Validate that `foldPos` is non-negative and use:

```java
foldPos <= targetType.parameterCount() - foldVals - foldArgs
```

**Testing:**

- Added tests for invalid positions and the valid boundary case where a zero-argument `void` combiner is folded at the end of the target parameter list.
- `make test TEST=test/jdk/java/lang/invoke`: 198 passed, 4 excluded, 0 failed.

---------
- [x] I confirm that I make this contribution in accordance with the [OpenJDK Interim AI Policy](https://openjdk.org/legal/ai).

---------
### Progress
- \[ \] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- \[x\] Change must not contain extraneous whitespace
- \[x\] Commit message must refer to an issue

### Issue
 * [JDK-8367022](https://bugs.openjdk.org/browse/JDK-8367022): MethodHandles.foldArguments handle negative pos (**Enhancement** - P4)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/32618/head:pull/32618` \
`$ git checkout pull/32618`

Update a local copy of the PR: \
`$ git checkout pull/32618` \
`$ git pull https://git.openjdk.org/jdk.git pull/32618/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 32618`

View PR using the GUI difftool: \
`$ git pr show -t 32618`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/32618.diff">https://git.openjdk.org/jdk/pull/32618.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/32618#issuecomment-5489874259)
</details>

### CODE REVIEW

File: test/jdk/java/lang/invoke/FoldTest.java [hunk 1]
```diff
@@ -92,6 +93,33 @@ public void testFoldArgumentsExample() throws Throwable {
...
+    @Test
+    public void testInvalidFoldPositions() {
+        MethodHandle voidCombiner = MethodHandles.empty(methodType(void.class, int.class));
+        int[] invalidPositions = {
+                -1,
+                Integer.MIN_VALUE,
+                Fold.MH_multer.type().parameterCount(),
+                Integer.MAX_VALUE
+        };
+        for (int pos : invalidPositions) {
```
Comment (JornVernee) [inline test/jdk/java/lang/invoke/FoldTest.java]: I suggest using `@ParameterizedTest` as that would still run all the cases instead of a failure.

**Review by JornVernee — COMMENTED** [review]
Thanks for fixing this! The changes look good, but this will also need a CSR.

/csr needed
